crates/macros/src/lib.rs

```diff
@@ -111,3 +115,34 @@ pub fn grug_export(_attr: TokenStream, mut item: TokenStream) -> TokenStream {
     item.extend(entry);
     item
 }
+
+#[proc_macro_attribute]
+pub fn index_list(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as ItemStruct);
+
+    let ty = Ident::new(&attr.to_string(), Span::call_site());
+    let struct_ty = input.ident.clone();
+
+    let names = input
+        .fields
+        .clone()
+        .into_iter()
+        .map(|e| {
+            let name = e.ident.unwrap();
+            quote! { &self.#name }
+        })
+        .collect::<Vec<_>>();
+
+    let expanded = quote! {
+        #input
+
+        impl cw_storage_plus::IndexList<#ty> for #struct_ty<'_> {
+            fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn cw_storage_plus::Index<#ty>> + '_> {
+                let v: Vec<&dyn cw_storage_plus::Index<#ty>> = vec![#(#names),*];
```

Comment:
Yeah. This macro has to be rewrite also because now the Indexes struct could have different Encoding schema.
See [here](https://github.com/left-curve/grug/blob/fd81a9030b317d8e2a142d3a4f300acd43a9dad5/crates/storage/src/index_map.rs#L230C1-L241)

Reply:
Even in a real-world environment, it makes no sense to define an `IndexMap` where the `Encoding` is changeable. In the contract, once decided, the schema will always be the same. So in the `macro`, we can omit the encoding and define the `IndexMap`as constant, where the `Encoding` must already be defined